cardano-testnet/src/Testnet/Defaults.hs

```diff
@@ -412,13 +413,15 @@ defaultByronProtocolParamsJsonValue =
 defaultShelleyGenesis
   :: UTCTime
   -> CardanoTestnetOptions
+  -> Maybe (Api.ShelleyGenesis StandardCrypto) -- ^ The file to use. If not provided, a default one is used.
+                                               --   Some fields are overridden by the accompanying 'CardanoTestnetOptions'
   -> Api.ShelleyGenesis StandardCrypto
-defaultShelleyGenesis startTime testnetOptions =
+defaultShelleyGenesis startTime testnetOptions mBase =
   let testnetMagic = cardanoTestnetMagic testnetOptions
       slotLength = cardanoSlotLength testnetOptions
       epochLength = cardanoEpochLength testnetOptions
       maxLovelaceLovelaceSupply = cardanoMaxSupply testnetOptions
-  in Api.shelleyGenesisDefaults
+  in (fromMaybe Api.shelleyGenesisDefaults mBase)
```

Comment:
So the problem here is if a user did provide a shelley genesis we would be overriding several fields.

Reply:
Yes, but this is mitigated by the additional documentation I have added in all the call chain about this.

Also, discussing a similar API with @andreabedini, this overriding is the default expectation of callers (not doing it would be the surprise!).

So I'd keep it like this.